Mathlib/Algebra/Quaternion.lean

```diff
@@ -1319,14 +1318,14 @@ alias rat_cast_imK := ratCast_imK
 alias coe_rat_cast := coe_ratCast
 
 instance instDivisionRing : DivisionRing ℍ[R] where
-  __ := Quaternion.instGroupWithZero
   __ := Quaternion.instRing
```

Comment:
I assume `fast_instance%` now also solves this?

Reply:
I would guess so also but I didn't test. Do you want to use it? I think there is diminishing returns here.

I also tried various mixtures of supplied instances but nothing seemed cleaner. 

Macro question is whether/when a synthesize instance should take precedent over (parts) of a supplied one.

Reply:
I think your change is ok. Using the elaborator had the advantage that changes to the hierarchy automatically result in the optimal choice here for downstream users, but the disadvantage of overhead to create the instance.